File: src/cli/XtPrint.cs
using System;
using System.Collections.Generic;

namespace Xt
{
    public static class XtPrint
    {
        public static string LevelToString(XtLevel level) => XtNative.StringFromUtf8(XtNative.XtPrintLevelToString(level));
        public static string CauseToString(XtCause cause) => XtNative.StringFromUtf8(XtNative.XtPrintCauseToString(cause));
        public static string SetupToString(XtSetup setup) => XtNative.StringFromUtf8(XtNative.XtPrintSetupToString(setup));
        public static string SystemToString(XtSystem system) => XtNative.StringFromUtf8(XtNative.XtPrintSystemToString(system));
        public static string SampleToString(XtSample sample) => XtNative.StringFromUtf8(XtNative.XtPrintSampleToString(sample));
        
        public static unsafe IList<string> CapabilitiesToString(XtCapabilities capabilities)
		{
            int i = 0;
            var result = new List<string>();
            var strings = (IntPtr*)XtNative.XtPrintCapabilitiesToString(capabilities);
            while (strings[i] != IntPtr.Zero)
                result.Add(XtNative.StringFromUtf8(strings[i++]));
            return result;
        }
    }
}

File: src/java/com/xtaudio/xt/XtPrint.java
package com.xtaudio.xt;

import com.sun.jna.Native;
import com.sun.jna.Pointer;
import java.util.ArrayList;
import java.util.List;

public final class XtPrint {

    private XtPrint() {
    }

    public static String levelToString(XtLevel level) {
        return XtNative.XtPrintLevelToString(level.ordinal());
    }

    public static String causeToString(XtCause cause) {
        return XtNative.XtPrintCauseToString(cause.ordinal());
    }

    public static String setupToString(XtSetup setup) {
        return XtNative.XtPrintSetupToString(setup.ordinal());
    }

    public static String systemToString(XtSystem system) {
        return XtNative.XtPrintSystemToString(system.ordinal() + 1);
    }

    public static String sampleToString(XtSample sample) {
        return XtNative.XtPrintSampleToString(sample.ordinal());
    }

    public static List<String> capabilitiesToString(int capabilities) {
        int i = 0;
        List<String> result = new ArrayList<>();
        Pointer strings = XtNative.XtPrintCapabilitiesToString(capabilities);
        while(strings.getPointer(i * Native.POINTER_SIZE) != null)
            result.add(strings.getPointer(i++ * Native.POINTER_SIZE).getString(0));
        return result;
    }
}